udata/core/site/models.py

```diff
@@ -28,6 +30,112 @@ class Site(WithMetrics, db.Document):
+        self.metrics["datasets"] = Dataset.objects.visible().count()
+        self.save()
+    
+    def count_resources(self):
+        self.metrics["resources"] = next(Dataset.objects.visible().aggregate(
+            {'$project': {'resources': 1}},
+            {'$unwind': '$resources' },
+            {'$group': {'_id': 'result', 'count': {'$sum': 1}}}
+        ), {}).get('count', 0)
+        self.save()
+
+    def count_reuses(self):
+        self.metrics["reuses"] = Reuse.objects.visible().count()
+        self.save()
+
+    def count_followers(self):
+        from udata.models import Follow
+        self.metrics["followers"] = Follow.objects(until=None).count()
+        self.save()
+    
+    def count_discussion(self):
+        from udata.models import Discussion
+        self.metrics["discussions"] = Discussion.objects.count()
+        self.save()
+    
+    def count_max_dataset_followers(self): 
+        dataset = (Dataset.objects(metrics__followers__gt=0).visible()
+                          .order_by('-metrics.followers').first())
+        self.metrics["max_dataset_followers"] = dataset.metrics['followers'] if dataset else 0
+        self.save()
+
+    def count_max_dataset_reuses(self):
+        dataset = (Dataset.objects(metrics__reuses__gt=0).visible()
+                        .order_by('-metrics.reuses').first())
+        self.metrics["max_dataset_reuses"] = dataset.metrics['reuses'] if dataset else 0
+        self.save()
+    
+    def count_max_reuse_datasets(self):
+        reuse = (Reuse.objects(metrics__datasets__gt=0).visible()
+                 .order_by('-metrics.datasets').first())
+        self.metrics["max_reuse_datasets"] = reuse.metrics['datasets'] if reuse else 0
+        self.save()
+
+    def count_max_reuse_followers(self):
+        reuse = (Reuse.objects(metrics__followers__gt=0).visible()
+                 .order_by('-metrics.followers').first())
+        self.metrics["max_reuse_followers"] = reuse.metrics['followers'] if reuse else 0
+        self.save()
+
+    def count_max_org_followers(self):
+        org = (Organization.objects(metrics__followers__gt=0).visible()
+               .order_by('-metrics.followers').first())
+        self.metrics["max_org_followers"] = org.metrics['followers'] if org else 0
+        self.save()
+
+    def count_max_org_reuses(self):
+        org = (Organization.objects(metrics__reuses__gt=0).visible()
+               .order_by('-metrics.reuses').first())
+        self.metrics["max_org_reuses"] = org.metrics['reuses'] if org else 0
+        self.save()
+
+    def count_max_org_datasets(self):
+        org = (Organization.objects(metrics__datasets__gt=0).visible()
+               .order_by('-metrics.datasets').first())
+        self.metrics["max_org_datasets"] = org.metrics['datasets'] if org else 0
+        self.save()
+    
+    @property
+    def get_max_metrics(self):
+        return {
+            "max_dataset_followers": self.metrics.get("max_dataset_followers", 0),
```

Comment:
nitpick: since those dict keys are the same than on `self.metrics`, maybe mutualise a list of keys.

Reply:
There's not exactly the same keys. It would means iterate on mutualised keys but filter to get discussions key for instance. Do you think of a clean way to do this? I think the code might just be less understandable. But it might also be the PTSD of the actual UData code being awfull to understand...